ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py

```diff
@@ -0,0 +1,202 @@
+from pathlib import Path
+import subprocess
+from argparse import ArgumentParser
+from typing import List, Union
+
+
+from ugvc.pipelines.single_cell_qc.collect_statistics import collect_statistics, extract_statistics_table
+from ugvc.pipelines.single_cell_qc.create_plots import (
+    cbc_umi_plot,
+    plot_mean_insert_quality_histogram,
+    plot_quality_per_position,
+    plot_insert_length_histogram,
+)
+from ugvc.pipelines.single_cell_qc.sc_qc_dataclasses import Inputs, OutputFiles, Thresholds
+from ugvc.utils.misc_utils import modify_jupyter_notebook_html
+
+
+def single_cell_qc(input_files: Inputs, output_path: str, thresholds: Thresholds):
+    """
+    Run single cell qc pipeline that collects statistics, prepares parameters for report and generates report
+
+    :param input_files: Inputs object with paths to input files
+    :param output_path: path to output directory
+    :param thresholds: Thresholds object with thresholds for qc
+    """
+
+    h5_file = collect_statistics(input_files, output_path)
+    extract_statistics_table(h5_file)
+
+    params, tmp_files = prepare_parameters_for_report(h5_file, thresholds, output_path)
+    output_report_html = generate_report(params, output_path, tmp_files=tmp_files)
+    # TODO: export h5_file and report html to papyrus
+
+
+def prepare_parameters_for_report(h5_file: Path, thresholds: Thresholds, output_path: str) -> Union[dict, List[Path]]:
```

Comment:
should be tuple and not a union
```suggestion
def prepare_parameters_for_report(h5_file: Path, thresholds: Thresholds, output_path: str) -> tuple[dict, list[Path]]:
```